tsconfig: "strict": true

error TS2345: Argument of type 'number | undefined' is not assignable to parameter of type 'number'.
  Type 'undefined' is not assignable to type 'number'.

► https://deno.land/x/dejs/dejs.ts:165:29

165     await statementBufWrite(buf.shift());
                                ~~~~~~~~~~~

--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,4 @@
-tsconfig.json
+/tsconfig.json
 node_modules
 package-lock.json
 package.json

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
 .PHONY: test
 
 test:
-	deno run --allow-read dejs_test.ts
+	deno -c ./testdata/tsconfig.json --allow-read dejs_test.ts

--- a/dejs.ts
+++ b/dejs.ts
@@ -128,14 +128,14 @@ export async function compile(reader: Reader): Promise<Template> {
         continue;
       }
       if (buf.length > 2) {
-        await statementBufWrite(buf.shift());
+        await statementBufWrite(buf.shift() as number);
       }
       continue;
     }
 
     // Finish current ReadMode
     if (buf[1] === Codes.Percent && buf[2] === Codes.End) {
-      statementBufWrite(buf.shift());
+      statementBufWrite(buf.shift() as number);
       buf.splice(0);
       // Don't execute if ReadMode is Comment.
       if (readMode !== ReadMode.Comment) {
@@ -157,12 +157,12 @@ export async function compile(reader: Reader): Promise<Template> {
       readMode = ReadMode.Normal;
       continue;
     }
-    await statementBufWrite(buf.shift());
+    await statementBufWrite(buf.shift() as number);
   }
 
   // Flush buffer
   while (buf.length > 0) {
-    await statementBufWrite(buf.shift());
+    await statementBufWrite(buf.shift() as number);
   }
   statements.push(`$$OUTPUT.push(\`${statementBuf.toString()}\`);`);
   statementBuf.reset();

--- a/testdata/tsconfig.json
+++ b/testdata/tsconfig.json
@@ -1,0 +1,9 @@
+{
+  "compilerOptions": {
+    "strict": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true
+  }
+}

